flash/text/generation/model.py

```diff
@@ -0,0 +1,98 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import warnings
+from typing import Optional, Union, Callable, Mapping, Type, Sequence, List, Dict, Any
+
+from pytorch_lightning import Callback
+from torchmetrics import Metric
+import torch
+from flash.core.utilities.imports import _TEXT_AVAILABLE
+
+from flash.core.model import Task
+from flash.text.ort_callback import ORTCallback
+
+if _TEXT_AVAILABLE:
+    from transformers import AutoModelWithLMHead, PreTrainedTokenizerBase
+    from transformers.models.gpt2.modeling_gpt2 import GPT2DoubleHeadsModelOutput
+
+else:
+    GPT2Model, PreTrainedTokenizerBase = None, None
+
+class TextGeneration(Task):
+    """
+
+    """
+    required_extras: str = "text"
+
+    def __init__(
+            self,
+            backbone: str = "gpt2",
+            loss_fn: Optional[Union[Callable, Mapping, Sequence]] = None,
+            optimizer: Type[torch.optim.Optimizer] = torch.optim.Adam,
+            metrics: Union[Metric, Callable, Mapping, Sequence, None] = None,
```

Comment:
Other parameters for the task have to be added, they were missing in other tasks previously as well which is now fixed.

```
        optimizer_kwargs: Optional[Dict[str, Any]] = None,
        scheduler: Optional[Union[Type[_LRScheduler], str, _LRScheduler]] = None,
        scheduler_kwargs: Optional[Dict[str, Any]] = None,
```

Reply:
Fixed